src/program.cpp

```diff
@@ -89,6 +89,73 @@ program::program(program&&) noexcept = default;
 program& program::operator=(program&&) noexcept = default;
 program::~program() noexcept                    = default;
 
+// copy constructor
+program::program(const program& p) { copy(p); }
+
+// copy assignment operator
+program& program::operator=(const program& p)
+{
+    if(this != &p)
+    {
+        copy(p);
```

Comment:
The assignment operator should use Copy-Swap:

```cpp
program& program::operator=(program p)
{
    std::swap(p.impl, impl);
}
```

Reply:
fixed